### PR TITLE
[React Native SDK] Fix: Remove device share if recovering private key fails

### PR DESCRIPTION
**Pull request by gregfromstl**
## PR-Codex overview
This PR introduces error handling improvements and a new function for managing device shares in the wallet system. It enhances user feedback for invalid device shares and ensures proper cleanup of device share data when errors occur.

### Detailed summary
- Added `INVALID_DEVICE_SHARE_MESSAGE` constant to provide a specific error message for invalid device shares.
- Implemented `removeDeviceShare` function to remove device shares based on `clientId`.
- Updated error handling in `getWalletPrivateKeyFromShares` to use the new error message.
- Modified `getAccountFromShares` to call `removeDeviceShare` when an invalid device state is detected.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: packages/thirdweb/src/wallets/in-app/native/helpers/wallet/sharded-wallet.ts [hunk 1]
```diff
@@ -67,6 +70,7 @@ export class ShardedWallet implements IWebWallet {
     return getExistingUserAccount({
       client: this.client,
       storage: this.storage,
+      storedToken: this.storedToken,
```
Comment (joaquim-verges) [inline packages/thirdweb/src/wallets/in-app/native/helpers/wallet/sharded-wallet.ts:73]: this feels scary. if you have it here in memory, can it ever be different form the one in local storage?

File: packages/thirdweb/src/wallets/in-app/native/helpers/wallet/retrieval.ts [hunk 2]
```diff
@@ -56,11 +63,34 @@ async function getWalletPrivateKeyFromShares(shares: string[]) {
...
+  const privateKey = await (async () => {
+    try {
+      return await getWalletPrivateKeyFromShares(shares);
+    } catch (e) {
+      // If the private key reconstruction fails, try to reset the device share
+      if (storedToken) {
+        await setUpShareForNewDevice({
+          client,
+          recoveryCode: await getRecoveryCode({
+            storedToken,
```
Comment (joaquim-verges) [inline packages/thirdweb/src/wallets/in-app/native/helpers/wallet/retrieval.ts:79]: you can't just grab this from local storage?

File: packages/thirdweb/src/wallets/in-app/native/helpers/storage/local.ts [hunk 3]
```diff
@@ -112,6 +112,22 @@ export async function setDeviceShare({
...
 
+export async function removeDeviceShare({
+  clientId,
+}: {
+  clientId: string;
+}): Promise<void> {
+  const userDetails = await getWalletUserDetails(clientId);
+
+  if (!userDetails) {
+    throw new Error("Missing wallet user ID");
```
Comment (joaquim-verges) [inline packages/thirdweb/src/wallets/in-app/native/helpers/storage/local.ts:123]: prob dont need to throw here? maybe just return early

File: packages/thirdweb/src/wallets/in-app/native/helpers/wallet/retrieval.ts [hunk 4]
```diff
@@ -58,9 +58,27 @@ async function getAccountFromShares(args: {
...
   const { client, shares } = args;
+  const privateKey = await (async () => {
+    try {
+      return await getWalletPrivateKeyFromShares(shares);
+    } catch (e) {
+      // If the private key reconstruction fails, try to reset the device share and prompt the user to try again
+      // This can happen if a user's account has been migrated or otherwise modified in the backend to use a new wallet. In that case, we need to reset their device state to get a new share
+      if (
+        e instanceof Error &&
+        e.message === "Invalid private key reconstructed from shares"
```
Comment (joaquim-verges) [inline packages/thirdweb/src/wallets/in-app/native/helpers/wallet/retrieval.ts:69]: would be good to put this string in a const so we dont change it accidentally and forget to update here

File: packages/thirdweb/src/wallets/in-app/native/helpers/storage/local.ts (hunk 3 above)
Comment (gregfromstl) [reply packages/thirdweb/src/wallets/in-app/native/helpers/storage/local.ts:123]: We won't be able to remove the device share though (even though one could exist)

**Review by joaquim-verges — APPROVED** [review]

**Review by joaquim-verges — COMMENTED** [review]
lil changeset for this?

**Pull request merged**